File: frontend/src/types/business.ts
export interface Business {
  id: string;
  name: string;
  description?: string;
  address?: string;
  phone?: string;
  email?: string;
  business_type: string;
  is_active: boolean;
  created_at: string;
  updated_at?: string;
}

export interface BusinessCreate {
  name: string;
  description?: string;
  address?: string;
  phone?: string;
  email?: string;
  business_type?: string;
}

export interface Product {
  id: string;
  business_id: string;
  name: string;
  description?: string;
  price: number;
  category?: string;
  image_url?: string;
  is_available: boolean;
  created_at: string;
  updated_at?: string;
}

export interface ProductCreate {
  business_id: string;
  name: string;
  description?: string;
  price: number;
  category?: string;
  image_url?: string;
}

File: backend/app/schemas/business.py
from pydantic import BaseModel
from typing import Optional, List
from uuid import UUID
from datetime import datetime

class BusinessBase(BaseModel):
    name: str
    description: Optional[str] = None
    address: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    business_type: Optional[str] = "general"

class BusinessCreate(BusinessBase):
    pass

class BusinessUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    address: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    business_type: Optional[str] = None
    is_active: Optional[bool] = None

class BusinessInDBBase(BusinessBase):
    id: UUID
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class Business(BusinessInDBBase):
    pass

class BusinessInDB(BusinessInDBBase):
    pass